tests/c/icmp_ptr.newcg.c

```diff
@@ -2,6 +2,7 @@
 // Run-time:
 //   env-var: YKD_SERIALISE_COMPILATION=1
 //   env-var: YKD_LOG_JITSTATE=-
+//   env-var: YKD_LOG_IR=-:jit-asm
```

Comment:
Last I checked, you can't use `jit-asm` in tests because it only works in debug builds.

Reply:
You can, and I use it frequently!

Reply:
You could probably remove it, since you aren't matching any asm?

Reply:
Huh, yes, good point! I guess I used it when debugging the asm. Fixed in 4cb0eab3364abebd21fb63c03b229ec154fcc8e2.

Reply:
Thanks.

I also check the disassembly routine. It's marked:
```
#[cfg(any(debug_assertions, test))]
```

I think here `test` is for unit tests, not for lang tests, so I doubt it would have worked anyway. The bookkeeping for printing matchable asm output is expensive, so it's disabled for release builds.

Reply:
I wouldn't expect it to work with `cargo test --release` but it does work with `cargo test` (I used this a *lot* during the development of this PR).

I think there is an interesting question about whether we want asm output in some way for release mode, because that tests/shows something quite different than debug mode. I'm not sure what to do there, and fortunately it's not question we need to answer right at this moment, so we can punt for a bit.